scalyr_agent/monitor_utils/k8s.py

```diff
@@ -566,42 +580,209 @@ def process_object( self, obj ):
-        self._pods = _K8sCache( logger, k8s, self._pod_processor, 'Pod',
-                               filter=self._node_filter )
+        self._pod_processor = PodProcessor( self._controllers )
+        self._pods = _K8sCache( self._pod_processor, 'Pod' )
 
         self._cluster_name = None
-        self._cache_expiry_secs = cache_expiry_secs
-        self._cache_purge_secs = cache_purge_secs
         self._last_full_update = time.time() - cache_expiry_secs - 1
 
-        self._lock = threading.Lock()
         self._initialized = False
 
-        self._thread = StoppableThread( target=self.update_cache, name="K8S Cache" )
+        self._thread = None
+
+        if start_caching:
+            self.start()
+
+    def _configure( self, old_config, new_config ):
+        """
+        Configures the cache based on any changes in the configuration
+        @param old_config: the old configuration
+        @param new_config: the new configuration
+        @type old_config: _CacheConfig
+        @type new_config: _CacheConfig
+        @return: True if the config changed, False otherwise
+        """
+
+        changed = False
+
+        # get old state values
+        old_k8s = None
+        old_namespace_filter = None
+        old_node_filter = None
+        old_config_update_count = 0
+        self._lock.acquire()
+        try:
+            old_k8s = self._k8s
+            old_namespace_filter = self._namespace_filter
+            old_node_filter = self._node_filter
+            old_config_update_count = self._config_update_count
+        finally:
+            self._lock.release()
+
+        # create a new k8s api object if we need one
+        k8s = old_k8s
+        if k8s is None or old_config.need_new_k8s_object( new_config ):
+            changed = True
+            if new_config.verify_api_queries:
+                k8s = KubernetesApi(k8s_api_url=new_config.api_url)
+            else:
+                k8s = KubernetesApi( ca_file=None, k8s_api_url=new_config.k8s_api_url)
+
+        # create new filters if we need them
+        namespace_filter = old_namespace_filter
+        node_filter = old_node_filter
+        if old_config.need_new_filters( new_config ):
+            changed = True
+            namespace_filter = self._build_namespace_filter( new_config.namespaces_to_ignore )
+            node_filter = self._build_node_filter( k8s, namespace_filter )
+
+        #see if expiry or purge times have changed
+        if old_config.cache_expiry_secs != new_config.cache_expiry_secs or old_config.cache_purge_secs != new_config.cache_purge_secs:
+            changed = True
+
+        # update with new values
+        self._lock.acquire()
+        try:
+            # if config update count has changed, it means a newer config change
+            # came through on another thread before we finished this call, so only make our changes
+            # if the update counts are the same
+            if self._config_update_count != old_config_update_count:
```

Comment:
Replace this with `if self._config is not new_config:  return False`.

Reply:
Done.